index.js

```diff
@@ -61,6 +61,8 @@ exports.parse = function(str){
     var parts = pairs[i].split('=');
     var key = decode(parts[0]);
     var m;
+    // Sanitize keys to uppercase to mitigate client-side prototype pollution
+    key && ["__proto__", "constructor", "prototype"].includes(key.toLowerCase()) ? key.toUpperCase() : key
```

Comment:
should we be setting `key` as a receiver here? I don't think the functions augment the variable.

```
key = 'hello'
'hello'
key.toUpperCase()
'HELLO'
key
'hello'
```

Reply:
Great catch — `toUpperCase()` returns a value, doesn't mutate in place.

Reply:
Thanks!  Implemented ^^